app/src/main/java/com/google/jetpackcamera/ui/JcaApp.kt

```diff
@@ -38,30 +41,63 @@ import com.google.jetpackcamera.ui.Routes.SETTINGS_ROUTE
 @OptIn(ExperimentalPermissionsApi::class)
 @Composable
 fun JcaApp(
+    openAppSettings: () -> Unit,
     /*TODO(b/306236646): remove after still capture*/
     previewMode: PreviewMode,
     onPreviewViewModel: (PreviewViewModel) -> Unit,
     onRequestWindowColorMode: (Int) -> Unit,
     modifier: Modifier = Modifier
 ) {
-    val permissionState =
-        rememberPermissionState(permission = Manifest.permission.CAMERA)
+    val cameraPermissionState = rememberPermissionState(permission = Manifest.permission.CAMERA)
+    val permissionStates = rememberMultiplePermissionsState(
+        permissions = listOf(
+            Manifest.permission.CAMERA,
+            Manifest.permission.RECORD_AUDIO
+        )
+    )
 
-    if (permissionState.status.isGranted) {
+    // should show rationale means a setting has been viewed and denied
+    if (cameraPermissionState.status.isGranted && (
+            permissionStates.allPermissionsGranted ||
+                permissionStates.shouldShowRationale
+            )
+    ) {
         JetpackCameraNavHost(
             onPreviewViewModel = onPreviewViewModel,
             previewMode = previewMode,
             onRequestWindowColorMode = onRequestWindowColorMode,
             modifier = modifier
         )
     } else {
-        CameraPermission(
+        // you'll have the option to go through camera and all other optional permissions
+        PermissionsScreen(
             modifier = modifier.fillMaxSize(),
-            cameraPermissionState = permissionState
+            permissionEnums = getUnGrantedPermissions(permissionStates),
+            openAppSettings = openAppSettings
         )
     }
 }
 
+@OptIn(ExperimentalPermissionsApi::class)
+private fun getUnGrantedPermissions(
```

Comment:
yeah that was a placeholder. can't use the term revoked or refused because it actually avoids revoked permissions. 
it only grabs permissions that are required and haven't been granted (camera)
Or Permissions that have been neither denied nor granted (first time the app is opened, or if the permission is set to ask every time)

I may just call it `getRequestablePermissions` and describe the rationale in a kdocs